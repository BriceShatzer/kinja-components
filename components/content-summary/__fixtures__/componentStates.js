/* @flow */

import type { SummaryProps } from '../content-summary';
import { createTypedTagDataId } from 'kinja-magma/models/Id';
import SimpleImage from 'kinja-magma/models/SimpleImage';

type ComponentStates = {
	[string]: {
		name: string,
		props: SummaryProps
	}
}

const defaultSummary = `
Are you a server's worst nightmare without even knowing it? We're here to help.
[The Salty Waitress](//thetakeout.com/c/the-salty-waitress) is The Takeout's advice column from a real-life waitress that
will teach you how not to behave like a garbage person while dining outâ€”and maybe
in real life.
`;

const image = new SimpleImage({
	id: 'vwppylx89l4ubmp3xcjc',
	format: 'jpg'
});

export const componentStates: ComponentStates = {
	storyTypeBoilerplateWithoutImage: {
		name: 'Story Type Boilerplate',
		props: {
			title: 'The Salty Waitress',
			summary: defaultSummary,
			blogId: 4,
			canonical: 'the-salty-waitress',
			canonicalHost: 'thetakeout.com',
			contentType: 'Standard',
			category: null,
			ga: () => {}
		}
	},
	storyTypeBoilerplateWithImage: {
		name: 'Story Type Boilerplate w/ Image',
		props: {
			title: 'The Salty Waitress',
			image,
			summary: defaultSummary,
			blogId: 4,
			canonical: 'the-salty-waitress',
			canonicalHost: 'thetakeout.com',
			contentType: 'Standard',
			category: null,
			ga: () => {}
		}
	},
	storyTypeBoilerplateWithImageAndOnlyPrev: {
		name: 'Story Type Boilerplate w/ Image & Previous Pagination',
		props: {
			title: 'The Salty Waitress',
			image,
			summary: defaultSummary,
			blogId: 4,
			canonical: 'the-salty-waitress',
			canonicalHost: 'thetakeout.com',
			contentType: 'Standard',
			prevPermalink: 'https://tv.avclub.com/the-expanse-slows-down-and-looks-to-the-past-1826634833',
			category: null,
			ga: () => {}
		}
	},
	storyTypeBoilerplateWithImageWithPagination: {
		name: 'Story Type Boilerplate w/ Image & w/ Pagination',
		props: {
			title: 'The Salty Waitress',
			image,
			summary: defaultSummary,
			blogId: 4,
			canonical: 'the-salty-waitress',
			canonicalHost: 'thetakeout.com',
			contentType: 'Standard',
			nextPermalink: 'https://tv.avclub.com/the-expanse-slows-down-and-looks-to-the-past-1826634833',
			prevPermalink: 'https://tv.avclub.com/the-expanse-slows-down-and-looks-to-the-past-1826634833',
			category: null,
			ga: () => {}
		}
	},
	storyTypeBoilerplateWithImageWithNonStandardContentType: {
		name: 'Story Type Boilerplate w/ Image & w/ Non-Standard Content Type',
		props: {
			title: 'The Salty Waitress',
			image,
			summary: defaultSummary,
			blogId: 4,
			canonical: 'the-salty-waitress',
			canonicalHost: 'thetakeout.com',
			contentType: 'Video',
			category: {
				canonicalName: 'reviews',
				valueDisplay: 'Reviews',
				id: createTypedTagDataId(0)
			},
			nextPermalink: 'https://tv.avclub.com/the-expanse-slows-down-and-looks-to-the-past-1826634833',
			prevPermalink: 'https://tv.avclub.com/the-expanse-slows-down-and-looks-to-the-past-1826634833',
			ga: () => {}
		}
	}
};
