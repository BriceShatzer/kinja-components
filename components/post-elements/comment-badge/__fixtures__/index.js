// @flow

import type { CommentBadgeProps } from '../comment-badge';

type States = {
	[string]: {
		name: string,
		props: CommentBadgeProps
	}
}

export const componentStates: States = {
	zeroCount: {
		name: 'renders a comment icon without a count when zero',
		props: {}
	},
	positiveCount: {
		name: 'renders a comment icon with a count of 1K when `count` is 1000',
		props: {
			count: 1000
		}
	}
};
