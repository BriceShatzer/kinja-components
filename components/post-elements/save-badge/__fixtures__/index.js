// @flow

import type { SaveBadgeProps } from '../save-badge';

type States = {
	[string]: {
		name: string,
		props: SaveBadgeProps
	}
}

export const componentStates: States = {
	zeroCount: {
		name: 'renders only an icon with "Save" when `count` is zero',
		props: {
			saveCount: 0
		}
	},
	nonZeroCount: {
		name: 'renders a number when `count` is above zero',
		props: {
			saveCount: 666
		}
	},
	isActive: {
		name: 'renders a filled icon when `isActive` is true',
		props: {
			isActive: true
		}
	},
	isNotActive: {
		name: 'renders an unfilled icon when `isActive` is false',
		props: {
			isActive: false
		}
	},
	noLabelOrCount: {
		name: 'renders only an icon with "Save," when props are empty',
		props: {}
	},
	iconOnly: {
		name: 'renders only an icon if `iconOnly` is on',
		props: {
			saveCount: 666,
			label: 'Test',
			iconOnly: true
		}
	},
	labelSupercedesCount: {
		name: 'renders a count over a label when both have positive values',
		props: {
			saveCount: 666,
			label: 'Test'
		}
	},
	countIs1e2: {
		name: 'renders 999 when count is 999',
		props: {
			saveCount: 999
		}
	},
	countIs1e3: {
		name: 'renders 1K when count is 1,000',
		props: {
			saveCount: 1e3
		}
	},
	countIs1e4: {
		name: 'renders 10K when count is 10,000',
		props: {
			saveCount: 1e4
		}
	},
	countIs1e5: {
		name: 'renders 100K when count is 100,000',
		props: {
			saveCount: 1e5
		}
	},
	countIs1e6: {
		name: 'renders `1M` with a count of 1,000,000',
		props: {
			saveCount: 1e6
		}
	}
};
