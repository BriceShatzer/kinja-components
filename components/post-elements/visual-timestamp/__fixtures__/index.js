// @flow

import type { VisualTimestampProps } from '../visual-timestamp';

type States = {
	[string]: {
		name: string,
		props: VisualTimestampProps
	}
}

export const componentStates: States = {
	live: {
		name: 'renders an animated live indicator',
		props: {
			isLive: true
		}
	},
	default: {
		name: 'renders a default clock icon for fmgNonSatire',
		props: {
			recircGroup: 'fmgNonSatire'
		}
	},
	editorialSplice: {
		name: 'renders an editorial splice icon for fmgNonSatire',
		props: {
			recircGroup: 'fmgNonSatire',
			isSpliced: true
		}
	},
	commerceSplice: {
		name: 'renders a deals icon for partnerEditorial',
		props: {
			recircGroup: 'partnerEditorial',
			isSpliced: true
		}
	},
	promotionsSplice: {
		name: 'renders a promotions icon for sponsored fmgBusiness',
		props: {
			recircGroup: 'fmgBusiness',
			isSponsored: true
		}
	},
	noProps: {
		name: 'renders without any props',
		props: {}
	},
	timestampFormatDefault: {
		name: 'renders "just now", relative time with default `timestampFormat`',
		props: {
			timestamp: Date.now()
		}
	}
};
