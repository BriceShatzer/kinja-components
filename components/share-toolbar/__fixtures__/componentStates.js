/* @flow */

import type {
	Props
} from '../share-toolbar';

type ComponentStates = {
	[string]: {
		name: string,
		props: Props
	}
}

export const componentStates: ComponentStates = {
	a: {
		name: 'Horizontal',
		props: {
			isVertical: false
		}
	}
};

