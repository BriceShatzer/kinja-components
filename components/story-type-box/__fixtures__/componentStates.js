/* @flow */

import type { StoryTypeBoxProps } from '../story-type-box';

type ComponentStates = {
    [string]: {
        name: string,
        props: StoryTypeBoxProps
    }
}

export const componentStates: ComponentStates = {
	textOnly: {
		name: 'Text only',
		props: {
			title: 'Ali\'s Top Reviews',
			description: 'This a page dedicated to everything that Ali loves the most in life.'
		}
	},
	textAndImage: {
		name: 'Text and image',
		props: {
			title: 'Ali\'s Top Reviews',
			description: 'This a page dedicated to everything that Ali loves the most in life.',
			image: {
				id: 'dvgmz3utpyl8fhivjejt',
				format: 'jpg'
			}
		}
	},
	textImageAndLabel: {
		name: 'Text, image and label',
		props: {
			title: 'Dogs',
			description: 'All About Good Doggos.',
			image: {
				id: 'cclianporp7wwg829w3g',
				format: 'jpg'
			},
			label: 'Ali\'s Top Reviews'
		}
	},
	titleAndImage: {
		name: 'Title and image',
		props: {
			title: 'Ali\'s Top Reviews',
			image: {
				id: 'dvgmz3utpyl8fhivjejt',
				format: 'jpg'
			}
		}
	},
	titleAndCategoryLabel: {
		name: 'Title and category label',
		props: {
			label: 'Ali\'s Top Reviews',
			title: 'Dogs'
		}
	}
};
