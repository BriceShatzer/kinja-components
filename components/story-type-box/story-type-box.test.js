/* @flow */

import * as React from 'react';
import { mount } from 'enzyme';

import StoryTypeBox from './story-type-box';

import { componentStates } from './__fixtures__/componentStates';

describe('<StoryTypeBox />', () => {

	it('should render as a box with title and description only', () => {
		const wrapper = mount(<StoryTypeBox {...componentStates.textOnly.props} />);
		expect(wrapper.render()).toMatchSnapshot();
	});

	it('should render as a box with title, description and image', () => {
		const wrapper = mount(<StoryTypeBox {...componentStates.textAndImage.props} />);
		expect(wrapper.render()).toMatchSnapshot();
	});

	it('should render as a box with title and image only', () => {
		const wrapper = mount(<StoryTypeBox {...componentStates.titleAndImage.props} />);
		expect(wrapper.render()).toMatchSnapshot();
	});

	it('should not render as a box, with title and category label only', () => {
		const wrapper = mount(<StoryTypeBox {...componentStates.titleAndCategoryLabel.props} />);
		expect(wrapper.render()).toMatchSnapshot();
	});

});
