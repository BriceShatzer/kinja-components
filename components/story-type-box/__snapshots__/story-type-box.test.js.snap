// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoryTypeBox /> should not render as a box, with title and category label only 1`] = `
.c3 {
  font-family: ProximaNovaCond,sans-serif;
  font-size: 14px;
  font-weight: bold;
  text-transform: uppercase;
  color: #0A7BC2;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid #0A7BC2;
  font-weight: normal;
  margin-right: 8px;
  padding: 5px 8px 3px;
  font-size: 14px;
  line-height: 16px;
}

.c3:hover {
  color: #085c91;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 1.125rem 0;
}

.c1 {
  padding: 0 1.125rem;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c2 {
  margin-bottom: 15px;
}

.c4 {
  margin: 0;
  font-family: ProximaNovaCond,sans-serif;
  font-weight: bold;
}

.c5 {
  color: #666;
  font-family: ElizabethSerif,Georgia,serif;
  font-size: 14px;
  line-height: 1.5;
  margin-top: 0.5rem;
}

@media only screen and (max-width:37.31em) {
  .c4 {
    font-size: 1.125rem;
  }
}

@media only screen and (min-width:37.37em) {
  .c4 {
    font-size: 1.75rem;
  }
}

@media only screen and (max-width:37.31em) {
  .c5 {
    margin-bottom: 1.5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        Ali's Top Reviews
      </span>
    </div>
    <h2
      class="c4"
    >
      Dogs
    </h2>
    <div
      class="c5"
    />
  </div>
</div>
`;

exports[`<StoryTypeBox /> should render as a box with title and description only 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 1.125rem 0;
  margin-bottom: 2.5rem;
  background-color: #F5F5F5;
}

.c1 {
  padding: 0 1.125rem;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c2 {
  margin: 0;
  font-family: ProximaNovaCond,sans-serif;
  font-weight: bold;
}

.c3 {
  color: #666;
  font-family: ElizabethSerif,Georgia,serif;
  font-size: 14px;
  line-height: 1.5;
  margin-top: 0.5rem;
}

@media only screen and (max-width:37.31em) {
  .c2 {
    font-size: 1.125rem;
  }
}

@media only screen and (min-width:37.37em) {
  .c2 {
    font-size: 1.75rem;
  }
}

@media only screen and (max-width:37.31em) {
  .c3 {
    margin-bottom: 1.5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Ali's Top Reviews
    </h2>
    <div
      class="c3"
    >
      This a page dedicated to everything that Ali loves the most in life.
    </div>
  </div>
</div>
`;

exports[`<StoryTypeBox /> should render as a box with title and image only 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 1.125rem 0;
  margin-bottom: 2.5rem;
  background-color: #F5F5F5;
}

.c1 {
  padding: 0 1.125rem;
}

.c2 {
  margin: 0;
  font-family: ProximaNovaCond,sans-serif;
  font-weight: bold;
}

.c3 {
  color: #666;
  font-family: ElizabethSerif,Georgia,serif;
  font-size: 14px;
  line-height: 1.5;
  margin-top: 0.5rem;
}

.c4 {
  display: block;
  width: 100%;
}

@media only screen and (max-width:37.31em) {
  .c1 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
  }
}

@media only screen and (min-width:37.37em) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media only screen and (max-width:37.31em) {
  .c2 {
    font-size: 1.125rem;
  }
}

@media only screen and (min-width:37.37em) {
  .c2 {
    font-size: 1.75rem;
  }
}

@media only screen and (max-width:37.31em) {
  .c3 {
    margin-bottom: 1.5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Ali's Top Reviews
    </h2>
    <div
      class="c3"
    />
  </div>
  <div
    class="c1"
  >
    <img
      class="c4"
      src="https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_180,q_80,w_320/dvgmz3utpyl8fhivjejt.jpg"
    />
  </div>
</div>
`;

exports[`<StoryTypeBox /> should render as a box with title, description and image 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 1.125rem 0;
  margin-bottom: 2.5rem;
  background-color: #F5F5F5;
}

.c1 {
  padding: 0 1.125rem;
}

.c2 {
  margin: 0;
  font-family: ProximaNovaCond,sans-serif;
  font-weight: bold;
}

.c3 {
  color: #666;
  font-family: ElizabethSerif,Georgia,serif;
  font-size: 14px;
  line-height: 1.5;
  margin-top: 0.5rem;
}

.c4 {
  display: block;
  width: 100%;
}

@media only screen and (max-width:37.31em) {
  .c1 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
  }
}

@media only screen and (min-width:37.37em) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media only screen and (max-width:37.31em) {
  .c2 {
    font-size: 1.125rem;
  }
}

@media only screen and (min-width:37.37em) {
  .c2 {
    font-size: 1.75rem;
  }
}

@media only screen and (max-width:37.31em) {
  .c3 {
    margin-bottom: 1.5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Ali's Top Reviews
    </h2>
    <div
      class="c3"
    >
      This a page dedicated to everything that Ali loves the most in life.
    </div>
  </div>
  <div
    class="c1"
  >
    <img
      class="c4"
      src="https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_180,q_80,w_320/dvgmz3utpyl8fhivjejt.jpg"
    />
  </div>
</div>
`;
